src/shared/ReactFiberComponentTreeHook.js

```diff
@@ -22,18 +22,50 @@ var {
 var describeComponentFrame = require('describeComponentFrame');
 var getComponentName = require('getComponentName');
 
+import type {Source} from 'ReactElementType';
 import type {Fiber} from 'ReactFiber';
 
+export type StackFrame = {
+  name: string | null,
+  ownerName: string | null,
+  source: Source | null,
+};
+
+function createStackFrame(fiber: Fiber): StackFrame {
+  switch (fiber.tag) {
+    case IndeterminateComponent:
+    case FunctionalComponent:
+    case ClassComponent:
+    case HostComponent:
+      const name = getComponentName(fiber);
+      const owner = fiber._debugOwner;
```

Comment:
These types are not sound. This will be `undefined` in PROD. All these accesses should be wrapped in DEV blocks.